### PR TITLE
[batch2] fix handling of invalid secrets and sas

### PR DESCRIPTION
**Pull request by jigold**

### CODE REVIEW

File: batch2/batch/batch.py [hunk 1]
```diff
@@ -13,6 +14,18 @@
...
 
+class KubernetesResourceError(Exception):
+    def __init__(self, resource_type, name, namespace):
+        super().__init__()
+        self.resource_type = resource_type
+        self.name = name
+        self.namespace = namespace
+
+    def __str__(self):
+        return f"could not get {self.resource_type} with name '{self.name}' in " \
```
Comment (cseed) [inline batch2/batch/batch.py]: Name is probably the most important here, I'd probably rephrase as "could not get resource {namespace}/{name} of kind {kind}".

File: batch2/batch/batch.py [hunk 2]
```diff
@@ -207,6 +220,28 @@ def getopt(obj, attr):
...
 
+async def get_k8s_secret(k8s_client, name, namespace):
+    try:
+        return await k8s_client.read_namespaced_secret(
+            name, namespace,
+            _request_timeout=KUBERNETES_TIMEOUT_IN_SECONDS)
+    except kube.client.rest.ApiException:
+        raise KubernetesResourceError('secret', name, namespace)
+    except:
+        raise
```
Comment (cseed) [inline batch2/batch/batch.py]: You don't need this.

File: batch2/batch/batch.py [hunk 3]
```diff
@@ -207,6 +220,28 @@ def getopt(obj, attr):
...
+
+async def get_k8s_service_account(k8s_client, name, namespace):
+    try:
+        return await k8s_client.read_namespaced_service_account(
+            name, namespace,
+            _request_timeout=KUBERNETES_TIMEOUT_IN_SECONDS)
+    except kube.client.rest.ApiException:
+        raise KubernetesResourceError('service account', name, namespace)
+    except:
+        raise
```
Comment (cseed) [inline batch2/batch/batch.py]: Same.

File: batch2/batch/batch.py [hunk 4]
```diff
@@ -301,12 +330,26 @@ def getopt(obj, attr):
...
+    try:
+        body = await job_config(app, record)
+    except Exception as err:
+        status = {
+            'worker': None,
+            'batch_id': batch_id,
+            'job_id': job_id,
+            'user': record['user'],
+            'state': 'error',
+            'error': str(err),
```
Comment (cseed) [inline batch2/batch/batch.py]: This should be traceback.format_exc(err), it has much more information, including backtrace and the exception chain.

**Comment by jigold** [comment]
Ok. I was trying to hide the kubernetes error message from the users because I thought it might be confusing. But if you feel that's ok, then I'll make it just report the original error.

**Comment by cseed** [comment]
Great; I was debating whether to suggest that.  I think we'll need sanitize the docker and k8s issues eventually, but for now, the more information the better.

File: batch2/batch/batch.py [hunk 5]
```diff
@@ -3,6 +3,8 @@
 import asyncio
 import aiohttp
 import base64
+import traceback
+import kubernetes_asyncio as kube
```
Comment (cseed) [inline batch2/batch/batch.py]: The import is unused.

**Pull request merged**